core/src/codegen.rs

```diff
@@ -815,6 +815,17 @@ impl Code for Statement {
                 gen.write(body);
             },
 
+            Statement::Do {
+                ref test,
+                ref body
+            } => {
+                gen.write_min(b"do ", b"do ");
+                gen.write(body);
+                gen.write_min(b"while (", b"while (");
+                gen.write(test);
+                gen.write_min(b")", b")");
```

Comment:
```rust
gen.write_bytes(b");")
```

All the `write_min` above can be changed to `write_bytes`. Can you make a `do` statement without a block?